great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -617,6 +625,13 @@ def validations_store_name(self, value: str) -> None:
     def validations_store(self) -> ValidationsStore:
         return self.stores[self.validations_store_name]
 
+    @property
+    def validation_config_store_name(self) -> str:
+        return DataContextConfigDefaults.DEFAULT_VALIDATION_CONFIG_STORE_NAME.value
+
+    def validaiton_config_store(self) -> ValidationConfigStore:
+        return self.stores[self.validation_config_store_name]
```

Comment:
Wasn't sure about this but deciding to not make `validation_config_store_name` a context variable. I've wavered on it but I'm now confident that store naming just causes more harm than benefit.

An argument against this is that it is a deviation from the existing pattern and perhaps consistency is more important?

Reply:
Do we know this key will always be present?
What is the harm you see from making this a context variable? By context variable, do you mean a top level property?

Reply:
By adding it as a context variable, its another field that a user can modify.

These top level store names are dangerous in that they can enable the following:
```python
context.checkpoint_store_name = "my_nonexistent_store"
...

context.checkpoint_store # Raises an error
context.checkpoints.add(...) # Raises an error
...
```

Reply:
We allow customization of the names but don't automatically update the `stores` dict, resulting in the possibility of orphaning a store by losing its name.

Seems relatively unlikely since we don't advertise this much in docs/public API.

We should always have a `ValidationConfigStore` but the name may vary.